include/dlaf/communication/datatypes.h

```diff
@@ -10,22 +10,25 @@
 
 #pragma once
 
-#include "dlaf/mpi_header.h"
-
-#include <complex>
+#include "dlaf/communication/internal/datatypes.h"
```

Comment:
Do we really need another header? Let's just keep things in `dlaf/communication/datatypes.h`